### ISSUE TITLE
Performance Optimization: Use MinMaxScaler instead of manual normalization for better efficiency and scalability

### ISSUE THREAD

**Issue by SaFE-APIOpt** [algonell/IPOMiner#2]
https://github.com/algonell/ipo-miner/blob/429148c4d9bc4d6a6fa6d06179d1825baa7b26f4/Notebooks/ml.py#L29
Hi, thank you for the great project!
The current normalization function is implemented as:
```
def normalize(df):
    '''Returns normalized DataFrame'''
    return (df - df.min()) / (df.max() - df.min())
```
Suggested Replacement:
```
from sklearn.preprocessing import MinMaxScaler

def normalize(df):
    '''Returns normalized DataFrame using sklearn MinMaxScaler'''
    scaler = MinMaxScaler()
    return pd.DataFrame(scaler.fit_transform(df), columns=df.columns, index=df.index)
```
he manual implementation relies on pandas’ high-level operations, which involve multiple passes over the data and generate intermediate DataFrame objects during subtraction and division. Each operation (min(), max(), arithmetic) is dispatched separately, and pandas handles type checking, broadcasting, and alignment for each one. This results in increased memory allocation and redundant computation, especially on large datasets.

**Comment by algonell** [comment]
Hi, thanks for the suggestion, indeed it could be better.
